src/librustc_span/hygiene.rs

```diff
@@ -104,9 +104,19 @@ impl ExpnId {
         HygieneData::with(|data| data.expn_data(self).clone())
     }
 
+    #[inline]
+    pub fn closest_ast_or_macro(self) -> ClosestAstOrMacro {
```

Comment:
I think a comment would be good here, even if it just instructs people to visit the `ClosestAstOrMacro` type for more information